src/macaron/slsa_analyzer/package_registry/jfrog_maven_registry.py

```diff
@@ -0,0 +1,302 @@
+        """
+        logger.info(
+            "Retrieving latest version of Java package %s:%s.",
+            group_id,
+            artifact_id,
+        )
+
+        group_path = group_id.replace(".", "/")
+
+        # JFrog REST API documentaiton: https://jfrog.com/help/r/jfrog-rest-apis/artifacts-storage.
+        url = f"{self.repo_url}/{self.repo_key}/{group_path}/{artifact_id}/maven-metadata.xml"
+
+        try:
+            response = requests.get(
+                url,
+                timeout=defaults.getint("requests", "timeout", fallback=10),
+            )
+        except requests.exceptions.RequestException as error:
+            logger.debug(
+                "Failed to retrieve the latest version of Java package %s:%s: %s", group_id, artifact_id, error
+            )
+            return None
+
+        if response.status_code == 200:
+            xml_payload = response.text
+            xml_tree = fromstring(xml_payload)
+        else:
+            logger.debug(
+                "Error retrieving maven-metadata.xml file of Java package %s:%s:  got response with status code %d.",
+                group_id,
+                artifact_id,
+                response.status_code,
+            )
+            return None
+
+        try:
+            return str(xml_tree.find("versioning").find("latest").text).strip()
+        except AttributeError as error:
+            # This happens if the maven-metadata.xml file does not have the
+            # /versioning/latest tag.
+            logger.debug(
+                "Error reading latest version of java package %s:%s from maven-metadata.xml file: %s",
+                group_id,
+                artifact_id,
+                error,
+            )
+            return None
+
+    def fetch_downloadable_assets(
+        self,
+        group_id: str,
+        artifact_id: str,
+        version: str,
+        extensions: list[str],
+    ) -> list[JFrogMavenAsset]:
+        """Retrieve the provenance file names for a particular version of a Java package.
+
+        Parameters
+        ----------
+        group_id : str
+            The group id of the Maven package.
+        artifact_id : str
+            The artifact id of the Maven package.
+        version : str
+            The version of the Maven package.
+        extensions : list[str]
+            The list of asset extensions.
+
+        Returns
+        -------
+        list[str]
+            The list of asset names.
+        """
+        group_path = group_id.replace(".", "/")
+
+        # JFrog REST API documentaiton: https://jfrog.com/help/r/jfrog-rest-apis/artifacts-storage.
+        url = f"{self.repo_url}/api/storage/{self.repo_key}/{group_path}/{artifact_id}/{version}"
+
+        try:
+            response = requests.get(
+                url=url,
+                timeout=defaults.getint("requests", "timeout", fallback=10),
+            )
+        except requests.exceptions.RequestException as error:
+            logger.error(
```

Comment:
Change to `debug` level log.

Reply:
This has been fixed in the current version.